nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -126,28 +128,70 @@ BtNavigator::on_configure(const rclcpp_lifecycle::State & /*state*/)
   blackboard_->set<int>("number_recoveries", 0);  // NOLINT
 
   // Get the BT filename to use from the node parameter
-  std::string bt_xml_filename;
-  get_parameter("bt_xml_filename", bt_xml_filename);
+  get_parameter("bt_xml_filenames", bt_xml_filenames_);
+  get_parameter("bt_xml_ids", bt_xml_ids_);
+
+  if (bt_xml_filenames_.empty()) {
+    RCLCPP_ERROR(get_logger(), "Couldn't find any XML file");
+    return nav2_util::CallbackReturn::FAILURE;
+  }
+
+  if (!loadBehaviorTree(bt_xml_ids_[0])) {
+    RCLCPP_ERROR(get_logger(), "Error loading XML file: %s", bt_xml_filenames_[0].c_str());
+    return nav2_util::CallbackReturn::FAILURE;
+  }
+
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+bool
+BtNavigator::loadBehaviorTree(const std::string & bt_id)
+{
+  // Use previous BT if it is the existing one
+  if (current_bt_id_ == bt_id) {
+    return true;
+  }
+
+  // Get the BT XML file corresponding to id
+  auto bt_id_it = std::find(bt_xml_ids_.begin(), bt_xml_ids_.end(), bt_id);
+  if (bt_id_it == bt_xml_ids_.end()) {
+    RCLCPP_ERROR(get_logger(), "BT id not found: %s", bt_id.c_str());
+    return false;
+  }
+
+  auto index = bt_id_it - bt_xml_ids_.begin();
+  if (index >= static_cast<int64_t>(bt_xml_filenames_.size())) {
+    RCLCPP_ERROR(get_logger(), "BT XML file not found at index: %ld", index);
+    return false;
+  }
+
+  // Get the full path f the BT XML
+  auto pkgpath = ament_index_cpp::get_package_share_directory("nav2_bt_navigator");
```

Comment:
This is bad, we cannot promise that all behavior trees will be in this arbitrary package. The file parameters would have to be file paths or something that would let us get to them.

Reply:
Removed. It is necessary to specify BT with full path